presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -141,32 +140,32 @@
 public class PlanPrinter
 {
     private final StringBuilder output = new StringBuilder();
-    private final Metadata metadata;
```

Comment:
why?

Reply:
Is that because you need a transaction to work with `Metadata`?
However `com.facebook.presto.metadata.MetadataManager#getFunctionRegistry` do need transaction today.

Reply:
No.  Function registry is constant.  If we do add support for transactional CREATE FUNCTION, the we will need to rethink this.